### ISSUE TITLE
sklearn tests not running in automatic tests

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#2886]
There seems to be something wrong with scikit-learn in the automatic unit tests, with lots of tests being skipped. For example: https://github.com/hyperspy/hyperspy/runs/5217344984?check_suite_focus=true#step:8:1961

Skips around 70 tests due to it not finding `sklearn`.

However, it should be installed, based on the install command being `pip install --upgrade -e .'[all, tests]'`, and `setup.py` containing:

```python
extras_require = {
    # exclude scikit-learn==1.0 on macOS (wheels issue)
    # See https://github.com/scikit-learn/scikit-learn/pull/21227
    "learning": ['scikit-learn!=1.0;sys_platform=="darwin"'],
    ...
}
```

**Comment by ericpre** [comment]
The syntax to define dependency requirement was incorrectly used, I fixed it in #2887.

**Comment by ericpre** [comment]
Done in https://github.com/hyperspy/hyperspy/pull/2887.
